### PR TITLE
Add MDX to blog starter

### PR DESCRIPTION
**Pull request by thomaswangio**
## Description

It was a hassle for me to convert my blog, which was based on the `gatsby-blog-starter`, to include MDX, so this PR adds MDX to the official blog starter.  

## Related Issues

> My biggest thought is this: should we just convert the starter blog to MDX for everyone, so they get it by default?

Addresses part of https://github.com/gatsbyjs/gatsby/issues/14607

### CODE REVIEW

**Comment by marcysutton** [comment]
Thanks for working on this! I will take a closer look at it next week, but I wanted to chime in that we need to test this thoroughly for performance. We've heard a lot about this recently, and MDX may contribute to bigger bundles and heavier pages. I wanted to mention that now while I had a moment!

**Comment by thomaswangio** [comment]
@marcysutton okay! if it's too heavy to introduce to the starter then I'll close this and just focus on making the advanced tutorial without this :)

File: starters/blog/gatsby-config.js [hunk 1]
```diff
@@ -53,7 +60,57 @@ module.exports = {
         //trackingId: `ADD YOUR TRACKING ID HERE`,
       },
     },
-    `gatsby-plugin-feed`,
+    {
+      resolve: `gatsby-plugin-feed`,
+      options: {
+        query: `
```
Comment (muescha) [inline starters/blog/gatsby-config.js]: should it make sense to make a new `gatsby-plugin-feed-mdx` for this use case
(maybe use the `gatsby-plugin-feed` but overwrite somehow the default query)

Comment (muescha) [reply starters/blog/gatsby-config.js]: i think it can be useful for other mdx sites

Comment (thomaswangio) [reply starters/blog/gatsby-config.js]: @muescha good idea!

Comment (thomaswangio) [reply starters/blog/gatsby-config.js]: @muescha I made `gatsby-plugin-feed-mdx` – https://github.com/thomaswangio/gatsby-plugin-feed-mdx. will replace it here as well as in the blog post!

Comment (thomaswangio) [reply starters/blog/gatsby-config.js]: tested it here https://github.com/thomaswangio/gatsby-personal-starter-blog/commit/4daab2d1609c560b62a9580e9d4fe16b878c4fe8 and it works!

**Comment by thomaswangio** [comment]
Closing this PR, continuing in https://github.com/gatsbyjs/gatsby/pull/19580

File: starters/blog/gatsby-config.js (hunk 1 above)
Comment (thomaswangio) [reply starters/blog/gatsby-config.js]: New PR: https://github.com/gatsbyjs/gatsby/pull/19580

Comment (muescha) [reply starters/blog/gatsby-config.js]: A sidenote: You don't need to close a PR.  Just commit new or changed files to the same branch - then the changes will show up here. 

-> That is more convenient to review. Reviewer only need to check the new commits and not the hole PR

Comment (muescha) [reply starters/blog/gatsby-config.js]: I see you make a new PR because you removed the fork to early

**Pull request closed without merging**